### DISCUSSION TITLE
User defined folder tasks

### DISCUSSION THREAD

**Discussion by igordsm** [elementary/code#1134] [Ideas]
Other code editors/IDEs provide the ability to run user defined commands (like tasks in VSCode or RunConfigurations in IntelliJ). I would like to implement something like this for Code. I've been thinking about some implementation ideas and would like other people's opinion on this. 

- custom JSON format (like in VSCode)
- GUI configuration tools like Visual Studio and IntelliJ
- .tasks folder with shell scripts that can be called from code and run in terminal
- some sort of makefile support

When a task is executed a new Terminal window would be launched with the command running. 

I'm not a huge fan of GUI for this type of task, so I have a preference for the other alternatives. The last two are specially attractive to me, since they leverage existing Linux knowledge.

**Comment by jeremypw** [comment]
I would be :+1: for the third (tasks folder) option.  I currently use bash aliases for common tasks like building rebuilding and linting a meson or flatpak project but these could easily be made into shellscripts accessible from the GUI.

**Reply by meisenzahl** [reply]
Could be a follow-up PR to https://github.com/elementary/code/pull/940.

**Reply by igordsm** [reply]
@meisenzahl #940 is a quite a complex contribution. I like the panel with the commands' output on the bottom and would love to use it as well. `ProjectManager` signals are also useful. It seems that what I'm planning is a lot less complex, but I'll be sure to reach out when design and implementation choices get clearer.
